module #3. REST API Advanced/src/main/java/com/epam/esm/controller/OrderController.java

```diff
@@ -0,0 +1,117 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.assembler.OrderAssembler;
+import com.epam.esm.dto.MakeOrderDto;
+import com.epam.esm.dto.OrderDto;
+import com.epam.esm.entity.User;
+import com.epam.esm.service.OrderService;
+import org.springframework.data.domain.Page;
+import org.springframework.hateoas.Link;
+import org.springframework.hateoas.PagedModel;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import javax.validation.constraints.Min;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.methodOn;
+
+/**
+ * This class is used to implement controller logic for {@link OrderDto orders}.
+ */
+@RestController
+@RequestMapping("api/orders")
+@Validated
+public class OrderController {
+
+    private final OrderService orderService;
+    private final OrderAssembler orderAssembler;
+
+    public OrderController(OrderService orderService, OrderAssembler orderAssembler) {
+        this.orderService = orderService;
+        this.orderAssembler = orderAssembler;
+    }
+
+    /**
+     * Method to get {@link OrderDto order} details by id.
+     * @param id The id of the {@link OrderDto order}
+     * @return The {@link OrderDto order} details
+     */
+    @GetMapping("/{id}")
+    public OrderDto orderById(
```

Comment:
Maybe it would be better to use such names as getById, getAll, getAllByUserId, etc.